interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -136,30 +135,15 @@ class NodeCompiler(
             contextWithOnlyGlobalVariables.withVariable(VariableConstants.InputVariableName, Unknown, paramName = None)
           NodeCompilationResult(Map.empty, None, defaultCtx, error)
       }
-    case frag @ FragmentInputDefinition(id, params, _) =>
-      def withFragmentParameterErrors[T](
-          compiledObject: ValidatedNel[ProcessCompilationError, T],
-          validationContext: ValidationContext
-      ) = {
-        def paramValidationErrors = params.map { param =>
-          fragmentParameterValidator.validate(param, id, validationContext)
-        }.sequence
-
-        compiledObject.andThen(v =>
-          paramValidationErrors match {
-            case Invalid(errors) => Invalid(errors)
-            case Valid(_)        => Valid(v)
-          }
-        )
-      }
+    case frag @ FragmentInputDefinition(id, _, _) =>
+      val parameterDefinitions =
+        fragmentDefinitionExtractor.extractParametersDefinition(frag, contextWithOnlyGlobalVariables)
+      val variables: Map[String, TypingResult] = parameterDefinitions.value.map(a => a.name -> a.typ).toMap
+      val validationContext                    = contextWithOnlyGlobalVariables.copy(localVariables = variables)
 
-      definitions.getComponent(ComponentType.Source, id) match {
+      val compilationResult = definitions.getComponent(ComponentType.Source, id) match {
```

Comment:
We need to understand this part :) 
1. Why this code is looking for some componen't definition? Shouldn't be always `None` here? 
2. `StubbedFragmentInputTestSource` takes `fragmentDefinitionExtractor` which looks weird. We already have `parameterDefinitions`.

Reply:
1. I understand that the case with `None` handling is for ad-hoc tests but why this code also do something with Some(...) case? When it happens?

Reply:
@philemone can you help us to understand your code? :)

Reply:
i think I don't understand your question, the Some(...) case is for ad-hoc testing, the None is the "normal behaviour", when just compiling a fragment

Reply:
In "normal behaviour" you don't have any Source with id = fragment id in definition. Why would you have it? definitions are something that you can use when you referring to something e.g. you use source component "kafka" in your nodes so you have to check the definition to figure out if you are using it correctly.

Here we are compiling fragment so we can't take a look at where is this fragment definition because we are just trying to figure out this definition :)

Reply:
ahh I read it again. You're right!

Reply:
I'm glad you worked that out :slightly_smiling_face: